### DISCUSSION TITLE
error on running ld_clump locally

### DISCUSSION THREAD

**Discussion by gxouc** [MRCIEU/TwoSampleMR#431] [Q&A]
Hi, sometimes the server was fine one week ago, but recently the server is busy and always show the following message:
Clumping sJAd2e, 2 variants, using EUR population reference
Server code: 503; Server is possibly experiencing traffic, trying again...
Server code: 502; Server is possibly experiencing traffic, trying again...
Server code: 502; Server is possibly experiencing traffic, trying again...
Server code: 502; Server is possibly experiencing traffic, trying again...
Server code: 502; Server is possibly experiencing traffic, trying again...
Server code: 502; Server is possibly experiencing traffic, trying again...
Server error: 502
Failed to retrieve results from server. See error status message in the returned object and contact the developers if the problem persists.
Removing 2 of 2 variants due to LD with other variants or absence from LD reference panel

Thus I decided to run ld_clump locally as described on https://rdrr.io/github/MRCIEU/gwasglue/f/vignettes/mr.Rmd. I got an error Saying : Error in file(file, "rt") : cannot open the connection
Here is the code:   mr_dat_exp_clumped <- ieugwasr::ld_clump(mr_dat_exp, clump_r2 = 0.8, plink_bin=get_plink_binary(), bfile="C:/data_maf0.01_rs_ref",clump_kb=1000,clump_p=5E-2)
Here is the file of "mr_dat_exp":
          SNP pval.exposure beta.exposure chr.exposure pos.exposure other_allele.exposure effect_allele.exposure se.exposure
1 rs140121611         2e-06           3.3        chr15     75762711                     A                      G    0.649873
2 rs191669673         2e-06           3.3        chr15     75788091                     T                      C    0.649873
                            exposure mr_keep.exposure pval_origin.exposure id.exposure eaf.exposure  pval                                 id
1 Brain - Spinal cord (cervical c-1)             TRUE             reported      sJAd2e           NA 2e-06 Brain - Spinal cord (cervical c-1)
2 Brain - Spinal cord (cervical c-1)             TRUE             reported      sJAd2e           NA 2e-06 Brain - Spinal cord (cervical c-1)
         rsid
1 rs140121611
2 rs191669673

I also downloaded the plink, but the error popped out.

**Comment by fabianacz** [comment]
I am having the same issue

I am trying to perform clumping using the 1000 genomes as reference dataset.

Here is my code

clump_data(
  summaryset_orru_format,
  clump_kb = 10000,
  clump_r2 = 0.001,
  clump_p1 = 1,
  clump_p2 = 1,
  pop = "EUR"
) 

Error in api_query("ld/clump", query = list(rsid = dat[["rsid"]], pval = dat[["pval"]],  : 
The query to MR-Base exceeded 300 seconds and timed out. Please simplify the query


I tried another approach. Running local LD operations. 

# To get a path to plink you can do the following:

devtools::install_github("explodecomputer/genetics.binaRies")
plink <- genetics.binaRies::get_plink_binary()

# To get the same LD reference dataset that is used by the API, you can download it directly from here:
 # http://fileserve.mrcieu.ac.uk/ld/1kg.v3.tgz

and tried this

orru_clump <- ld_clump(
  dplyr::tibble(rsid = summaryset_orru$rsid, pval = summaryset_orru$p, id = summaryset_orru$id),
  plink_bin = plink,
  bfile = "1kg.v3" #this is in the same directory of my R project
)

The outcome is 

Clumping ebi-a-GCST90001942, 15141982 variants, using EUR population reference
PLINK v1.90b7 64-bit (16 Jan 2023)             www.cog-genomics.org/plink/1.9/
(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3
Logging to C:\Users\corsfmdg\AppData\Local\Temp\Rtmp63D8xl\file393034026c62.log.
Options in effect:
  --bfile 1kg.v3
  --clump C:\Users\corsfmdg\AppData\Local\Temp\Rtmp63D8xl\file393034026c62
  --clump-kb 10000
  --clump-p1 0.99
  --clump-r2 0.001
  --out C:\Users\corsfmdg\AppData\Local\Temp\Rtmp63D8xl\file393034026c62

65210 MB RAM detected; reserving 32605 MB for main workspace.
**Error: Failed to open 1kg.v3.bed.**
**Error in file(file, "rt") : cannot open the connection
In addition: Warning message:
In file(file, "rt") :
  cannot open file 'C:\Users\corsfmdg\AppData\Local\Temp\Rtmp63D8xl\file393034026c62.clumped': No such file or directory**
